design-reference/architecture/VersionAndGo/readme.md

```diff
@@ -0,0 +1,113 @@
+# Version and Go Technical Design Purpose
+
+This design addresses Cobalt's workflow to deploy individual application infrastructure using a Cobalt template.
+
+## Definitions
+ - `Engineering Team (Eng Team)` - DevOps or Operations team consisting of Infrastructure Administrators or similar roles.
+ - `Application Team (App Team)` - Software Developers with an application to deploy to Azure.
+ - `Development Repo (Dev Repo)` - Cobalt's repo hosted on Github at [https://github.com/microsoft/cobal](https://github.com/microsoft/cobal)
+ - `Engineering Repo (Eng Repo)` - Individual template repository managed by the `Engineering Team`
+ - `Application Repo (App Repo)` - Repository that contains the application code to be deployed and managed by the `Application Team`
+ - `Version` - all versioning schemes should use [SemVer](https://semver.org/) or variant. 
+
+## Version and Go Setup Description
+Our design starts after the initial Cobalt setup, where an individual template is cloned into an organizations source control (ref: [Setup Doc](https://github.com/microsoft/cobalt/blob/master/docs/GETTING_STARTED_APP_DEV.md)). An application development team is ready to start deploying their code to infrastructure built in Azure. Typically they would make a request to the engineering team. The Eng team would provide the App team with a link to the latest infra configuration files for a specific template. The App team downloads this tar/zip file and extracts the folder/files.
+
+```
+infra/
+ |_ Application.tf
+ |_ Application.yaml
+```
+
+These files should be derived from the template folder in dev repo and cloned into the eng repo. Proposed folder path/location is:
+
+```
+infra/
+  |_template
+      |_template_name
+          |_tests
+          |_config
+          |   |_ application.tf
+          |   |_ application.yml
+          |_template files...
+```
+
+ - `Appliation.tf` is the artifact that points to the template in the eng repo. The application team would need to fill this file out with the requested configuration. Example:
```

Comment:
I really like this approach as it minimizes the content that the application teams need to pull in as dependencies. This also makes it easier to understand which versions teams in the case that there are migration efforts to move to the latest infrastructure best practices within an organization.